src/pugixml.cpp

```diff
@@ -3479,11 +3491,21 @@ PUGI_IMPL_NS_BEGIN
 						}
 						else
 						{
-							PUGI_IMPL_PUSHNODE(node_pcdata); // Append a new node on the tree.
+							xml_node_struct* cursor_last_child = cursor->first_child ? cursor->first_child->prev_sibling_c + 0 : 0;
 
-							cursor->value = s; // Save the offset.
+							if(PUGI_IMPL_OPTSET(parse_merge_pcdata) && cursor_last_child && PUGI_IMPL_NODETYPE(cursor_last_child) == node_pcdata)
+							{
+								strconcat(cursor_last_child->value, s);
+								s = cursor_last_child->value;
```

Comment:
I believe this will lead to double unescaping in strconv_pcdata below, which, in addition to being suboptimal for performance, will not be correct for something like this:

```
<node>
&amp;lt; <!-- comment --> &amp;gt;
</node>
```

(the above should result in `&lt;  &gt;` after parsing, but will probably result in `<  &gt;`)